dashboard/app/models/user.rb

```diff
@@ -1060,7 +1054,7 @@ def next_unpassed_visible_progression_level(script)
     sl_level_ids = visible_sls.map(&:level_ids).flatten
 
     # Levels the user made progress in
-    ul_level_ids = user_levels_by_level(script).keys
+    ul_level_ids = Queries::UserProgress.user_levels_by_level(self, script).keys
```

Comment:
Seems like it might be cleaner to just rewrite this to use regular rails proceses;
```suggestion
    ul_level_ids = user_levels.where(script: script).pluck(:id)
```